third_party/terraform/tests/resource_storage_transfer_job_test.go

```diff
@@ -0,0 +1,165 @@
+			{
+				ResourceName:      "google_storage_transfer_job.transfer_job",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccStorageTransferJob_basic(getTestProjectFromEnv(), testDataSourceId, testUpdatedDataSinkId, testTransferJobId),
+				Check:  resource.TestCheckResourceAttr("google_storage_transfer_job.transfer_job.transfer_spec.0.gcs_data_sink.0", "bucket_name", testUpdatedDataSinkId),
+			},
+			{
+				ResourceName:      "google_storage_transfer_job.transfer_job",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccStorageTransferJob_basic(getTestProjectFromEnv(), testUpdatedDataSourceId, testUpdatedDataSinkId, testTransferJobId),
+				Check:  resource.TestCheckResourceAttr("google_storage_transfer_job.transfer_job.transfer_spec.0.gcs_data_source.0", "bucket_name", testUpdatedDataSinkId),
+			},
+			{
+				ResourceName:      "google_storage_transfer_job.transfer_job",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccStorageTransferJob_basic(project string, dataSourceId string, dataSinkId string, transferJobId string) string {
+	return fmt.Sprintf(`
+data "google_storage_transfer_project_service_account" "default" {
+  project       = "%s"
+}
+
+resource "google_storage_bucket" "data_source" {
+  name          = "test-data-source-bucket-%s"
+  project       = "%s"
+  force_destroy = true
+}
+
+resource "google_storage_bucket_iam_member" "data_source" {
+  bucket        = "${google_storage_bucket.data_source.name}"
+  role          = "roles/storage.admin"
+  member        = "serviceAccount:${data.google_storage_transfer_project_service_account.default.email}"
+
+  depends_on    = [
+    "google_storage_bucket.data_source",
+    "data.google_storage_transfer_project_service_account.default"
+  ]
+}
+
+resource "google_storage_bucket" "data_sink" {
+  name          = "test-data-sink-bucket-%s"
+  project       = "%s"
+  force_destroy = true
+}
+
+resource "google_storage_bucket_iam_member" "data_sink" {
+  bucket        = "${google_storage_bucket.data_sink.name}"
+  role          = "roles/storage.admin"
+  member        = "serviceAccount:${data.google_storage_transfer_project_service_account.default.email}"
+
+  depends_on    = [
+    "google_storage_bucket.data_sink",
+    "data.google_storage_transfer_project_service_account.default"
+  ]
+}
+
+resource "google_storage_transfer_job" "transfer_job" {
+	description	= "transfer-job-%s"
+	project     = "%s"
+
+	transfer_spec {
+		gcs_data_source {
+			bucket_name = "${google_storage_bucket.data_source.name}"
+		}
+		gcs_data_sink {
+			bucket_name = "${google_storage_bucket.data_sink.name}"
+		}
+	}
+
+	schedule {
+		schedule_start_date {
+			year	= 2018
+			month	= 10
+			day		= 1
+		}
+		schedule_end_date {
+			year	= 2019
+			month	= 10
+			day		= 1
+		}
+		start_time_of_day {
+			hours	= 0
+			minutes	= 30
+			seconds	= 0
+			nanos	= 0
+		}
+	}
+
+	depends_on = [
```

Comment:
For completeness, this is not redundant- Terraform doesn't know to wait for those IAM roles to be applied without this block.